polygeist/tools/cgeist/Test/Verification/booleanconversion.c

```diff
@@ -0,0 +1,72 @@
+// RUN: cgeist %s --function=* -S -O0 | FileCheck %s
+
+#include <stdbool.h>
+
+// CHECK-LABEL:   func.func @int_conversion(
+// CHECK-SAME:                              %[[VAL_0:.*]]: i32) -> i8
+// CHECK-NEXT:      %[[VAL_1:.*]] = arith.constant 0 : i32
+// CHECK-NEXT:      %[[VAL_2:.*]] = arith.cmpi ne, %[[VAL_0]], %[[VAL_1]] : i32
+// CHECK-NEXT:      %[[VAL_3:.*]] = arith.extui %[[VAL_2]] : i1 to i8
```

Comment:
Remove and return the i1 directly.

Reply:
We cannot do that, as that would require many changes. I'm gradually working on that. Will need to revamp how we translate types to MLIR.

Reply:
That's fine for this PR.  Now that you mention it I recall we have seen this issue before.  I just wanted to point it out because we will need to fix it in the future to achieve compatibility with code compiled by stock clang.